nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -69,11 +76,13 @@ async function makeBlock(proposer, number = TRANSACTIONS_PER_BLOCK) {
 
 /**
  * This function will make a block iff I am the proposer and there are enough
- * transactions in the database to assembke a block from. It loops until told to
+ * transactions in the database to assemble a block from. It loops until told to
  * stop making blocks. It is called from the 'main()' routine to start it, and
  * should not be called from anywhere else because we only want one instance ever
  */
 export async function conditionalMakeBlock(proposer) {
+  const stateContractInstance = await waitForContract(STATE_CONTRACT_NAME);
+  const stateContractAddress = await getContractAddress(STATE_CONTRACT_NAME);
```

Comment:
Just for later reviewing, all this pattern of getting stateContractInstances is happening all over the place... Wondering if we could abstract it somewhere and simply return an object like:

```
{
    "state": "0x1234",
    "proposers", "0x5678"
}
````

We could then call this func somewhere in the ```app.mjs``` file and reuse it.

For later, though :)